static/js/components/forms/B2BPurchaseForm.js

```diff
@@ -60,18 +67,28 @@ class B2BPurchaseForm extends React.Component<Props> {
         setFieldError("coupon", "No product selected")
         return
       }
+      let productId = values.product
+      if (isNaN(values.product)) {
```

Comment:
Just noticed the test coverage is a little low - can you add tests to cover both product id (numeric) and text_id (non-numeric) for this code here and some other sections that codecov indicates is missing coverage?